homeassistant/components/smlight/sensor.py

```diff
@@ -0,0 +1,166 @@
+"""Support for SLZB-06 sensors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, UnitOfInformation, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.dt import utcnow
+
+from .const import DOMAIN, SCAN_INTERVAL, SMLIGHT_SLZB_REBOOT_EVENT
+from .coordinator import SmData, SmDataUpdateCoordinator
+from .entity import SmEntity
+
+UPTIME_DEVIATION = 5  # seconds
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class SmSensorEntityDescription(SensorEntityDescription):
+    """Class describing SMLIGHT sensor entities."""
+
+    value_fn: Callable[[SmData], float | None] = lambda _: None
+
+
+SENSORS = [
+    SmSensorEntityDescription(
+        key="esp32_temperature",
```

Comment:
is there a reason why the first two `key` and `translation_keys` are different? Might lead to confusion

Reply:
translation keys got renamed to user friendly names, but must have missed changing keys.